### ISSUE TITLE
Feature request: Preprocess data to display missing data in graphs

### ISSUE THREAD

**Issue by SlimeDog** [Rsl1122/Plan-PlayerAnalytics#534]
When a server is down, no data are recorded (of course). Unfortunately, the x-axis of the current HighCharts graphs is simply a collection of ordered points, with no knowledge of the underlying time-series implicit in the data. So missing data are not displayed as null (holes in the series display) or zeros.

As I'm reading more on HighCharts forums, it looks like some preprocessing will be required to produce a complete time-series with nulls (ie., holes in the time series). Alternatively, perhaps there is a built-in time-series chart type.

It would be nice to make this a configurable option, with missing data displayed as either null (ie., no points) or zero, configurable per chart.

**Comment by SlimeDog** [comment]
On reflection, null is the correct default for missing data, not zero. There should just be holes in the graph.

**Comment by Rsl1122** [comment]
I tested with both null & 0 - zero was too misleading so only null option was added.

The setting increases page loading time (Due to increased amount of points) & gap data interval had to be limited to 30 minutes for large gaps.
It also causes HighCharts graph drawing optimization to have some issues with drawing short spikes (Such as a single session of 20 minutes) to become worse and thus is disabled by default.

![1](http://puu.sh/zFVRv/ac4f722133.png)
*A spike of 1 player that is unreadable due to zoom, zooming in displays the spike correctly, but it is invisible in the graph when viewing with a large zoom. This issue is unrelated to the gaps, but is made worse by it.*
![2](http://puu.sh/zFVSg/e5f205f36f.png)
*Performance chart with downtime*

Because I implemented it on the Abstract chart level, changing it to only appear on some charts might require more work & settings. I'll leave it like this for now.
